src/masternodes/poolpairs.h

```diff
@@ -280,7 +285,28 @@ class CPoolPairView : public virtual CStorageView
             CAmount distributedFeeA = 0;
             CAmount distributedFeeB = 0;
 
-            if (pool.totalLiquidity == 0 || (!pool.swapEvent && poolReward == 0)) {
+            auto rewards = GetPoolCustomReward(poolId);
+            bool payCustomRewards{false};
+
+            if (nHeight >= Params().GetConsensus().ClarkeQuayHeight && rewards && !rewards->balances.empty()) {
+                for (auto it = rewards->balances.cbegin(), next_it = it; it != rewards->balances.cend(); it = next_it) {
+                    ++next_it;
+
+                    // Get token balance
+                    const auto balance = onGetBalance(pool.ownerAddress, it->first).nValue;
+
+                    // Make there's enough to pay reward otherwise remove it
+                    if (balance < it->second) {
+                        rewards->balances.erase(it);
```

Comment:
I think this would be up to the pool owner to check and we may consider this not to actually be a problem but intentional, for example we no longer want to pay out that token so let it deplete or moved the token out of the owner address. If we do log this then we are logging for everyone, if we do log it then I would suggest adding it to a debug category to keep it out of regular logging.

Reply:
You can check owner is mine for wallet, then log? I think it can notable if you want to pay but you don't know you run out of money.

Reply:
That's going to create at least one log entry per block, we would not want to trigger it for each reward payment for sure. The pool owner can just run the following.

`getaccount poolOwner`

My take on it is that if it ran out of tokens then it was intentional, that's how you would stop payment. Let's get someone to take the deciding vote @uzyn :slightly_smiling_face:

Reply:
I'm OK with explanation. Another option is to update ppol custom reward by disable / erase that reward, then owner can enable it again when update its balance

Reply:
Did actually have this functionality at one point to remove token, '0@token' actually removed the token from rewards, however RPC call creates errors from DecodeAmounts so I removed it from RPC and ApplyUpdatePoolPairTx, did not make sense to support a feature that could not be accessed. Will restore and work round this DecodeAmounts error.